waspc/data/Generator/templates/sdk/wasp/server/auth/oauth/provider.ts

```diff
@@ -1,13 +1,14 @@
 import { OAuth2Provider, OAuth2ProviderWithPKCE } from "arctic";
 
 export function defineProvider<
-  OAuthClient extends OAuth2Provider | OAuth2ProviderWithPKCE
+  OAuthClient extends OAuth2Provider | OAuth2ProviderWithPKCE,
+  const Id extends string
```

Comment:
Adding the `const` type parameter makes it output a e.g. `{id: "discord"}` type instead of a `{id: string}` one. So then it typechecks correctly for the login provider actions.

Reply:
Nice improvement!